crates/ruma-state-res/src/event_auth.rs

```diff
@@ -515,48 +511,45 @@ fn valid_membership_change<E: Event>(
                 let allow = target_user_current_membership == MembershipState::Join
                     || target_user_current_membership == MembershipState::Invite;
                 if !allow {
-                    warn!(
-                        "Can't leave if not invited or joined\nCurrent user state: {:?}",
-                        target_user_membership_event_id,
-                    );
+                    warn!(?target_user_membership_event_id, "Can't leave if not invited or joined");
                 }
                 allow
             } else if !sender_is_joined
                 || target_user_current_membership == MembershipState::Ban
                     && sender_power.filter(|&p| p < &power_levels.ban).is_some()
             {
-                warn!("Can't kick if sender not joined or user is already banned\nCurrent user state: {:?}\nSender user state: {:?}",
-                    target_user_membership_event_id,
-                    sender_membership_event_id
+                warn!(
+                    ?target_user_membership_event_id,
+                    ?sender_membership_event_id,
+                    "Can't kick if sender not joined or user is already banned",
                 );
                 false
             } else {
                 let allow = sender_power.filter(|&p| p >= &power_levels.kick).is_some()
                     && target_power < sender_power;
                 if !allow {
-                    warn!("User does not have enough power to kick\nCurrent user state: {:?}\nPower levels: {:?}",
-                        target_user_membership_event_id,
-                        power_levels_event_id,
+                    warn!(
+                        ?target_user_membership_event_id,
+                        ?power_levels_event_id,
+                        "User does not have enough power to kick",
```

Comment:
What does this output look like, I'm fine with the change I just couldn't really find examples in the macro docs for tracing?

Reply:
Depends on the formatter. By the way, just found out about this, think we should be using it: https://docs.rs/tracing-subscriber/0.2.20/tracing_subscriber/fmt/struct.SubscriberBuilder.html#method.with_test_writer

Reply:
In the [`TestWriter`](https://docs.rs/tracing-subscriber/0.2.20/tracing_subscriber/fmt/writer/struct.TestWriter.html) docs it says

> Writing to io::stdout and io::stderr produces the same results as using libtest’s --nocapture option which may make the results look unreadable.

This is just saying if you have a bunch of `println` / `eprintln`s that would mess it up? We don't have any so it should be fine. The way to get the output would then be `RUST_LOG=whatever cargo test -- --show-output / --no-capture`?

Reply:
Using the `TestWriter` seems reasonable to me.